From: Purushotam Kumar <purushotam@ti.com>
Date: Fri, 18 Jul 2008 23:28:57 +0000 (-0700)
Subject: OMAP3:devices.c:Enabling 4-bit for SD card
X-Git-Url: http://www.sakoman.net/cgi-bin/gitweb.cgi?p=linux-omap-2.6.git;a=commitdiff_plain;h=6c4d34031c80ca4b50ffe73a4ef7fe197a760a60

OMAP3:devices.c:Enabling 4-bit for SD card

SD card was working in 1-bit mode.This patch will configure SD card in
4-bit mode and hence performance  will increase.

Signed-off-by: Purushotam Kumar <purushotam@ti.com>
Acked-by: Madhusudhan Chikkature Rajashekar <madhu.cr@ti.com>
---

diff --git a/arch/arm/plat-omap/devices.c b/arch/arm/plat-omap/devices.c
index b83f9a6..d0cfceb 100644
--- a/arch/arm/plat-omap/devices.c
+++ b/arch/arm/plat-omap/devices.c
@@ -296,13 +296,17 @@ static void __init omap_init_mmc(void)
 	mmc = &mmc_conf->mmc[0];
 
 	if (cpu_is_omap2430() || cpu_is_omap34xx()) {
-		if (mmc->enabled)
+		if (mmc->enabled) {
+			mmc1_data.conf = *mmc;
 			(void) platform_device_register(&mmc_omap_device1);
+		}
 
 #if defined(CONFIG_ARCH_OMAP2430) || defined(CONFIG_ARCH_OMAP34XX)
 		mmc = &mmc_conf->mmc[1];
-		if (mmc->enabled)
+		if (mmc->enabled) {
+			mmc2_data.conf = *mmc;
 			(void) platform_device_register(&mmc_omap_device2);
+		}
 #endif
 
 		return;
